### PR TITLE
Consume offer

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following commit bumps to the latest charm.v6 dep, which takes
the offer url from the juju package and puts it into the charm.v6
dep so that we can easily validate offer urls from the bundle data
perspective without having to duplicate code. With this in mind
we also change the juju code to use that new changes.

Drive by fixes include cleaning up whitespace on lines

## QA steps

```console
export JUJU_DEV_FEATURE_FLAGS=bundle-cmr
juju bootstrap lxd test --no-gui

juju deploy mysql
juju status --relations
```
You should see mysql in the relation section
```console
juju offer mysql:db
juju add-model other
juju switch other
juju deploy ./testcharms/charm-repo/bundle/wordpress-with-saas/bundle.yaml
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
!!build!!

**Comment by SimonRichardson** [comment]
https://trello.com/c/hdRcwK9W/1516-cmr-apply-consume-offer-change-from-bundlechanges-when-deploying-bundle

File: cmd/juju/application/bundle.go [hunk 1]
```diff
@@ -241,6 +244,13 @@ type bundleHandler struct {
 
 	// The UUID of the model where the bundle is about to be deployed.
 	targetModelUUID string
+
+	// Controller name required for consuming a offer when deploying a bundle.
+	controllerName string
+
+	// accountUser holds the user of the account associated with the
+	// current controller.
+	accountUser string
```
Comment (achilleasa) [inline cmd/juju/application/bundle.go]: Nice! I was looking for this one for my PR as well

File: cmd/juju/application/deploy.go [hunk 2]
```diff
@@ -71,6 +72,7 @@ type ApplicationAPI interface {
 	SetConstraints(application string, constraints constraints.Value) error
 	Update(apiparams.ApplicationUpdate) error
 	ScaleApplication(application.ScaleApplicationParams) (apiparams.ScaleApplicationResult, error)
+	Consume(arg crossmodel.ConsumeApplicationArgs) (string, error)
```
Comment (achilleasa) [inline cmd/juju/application/deploy.go]: It's interesting that this call is not part of the offer API

File: cmd/juju/application/deploy.go [hunk 3]
```diff
@@ -1363,6 +1375,8 @@ func (c *DeployCommand) maybeReadLocalBundle(ctx *cmd.Context) (deployFn, error)
 			bundleMachines:      c.BundleMachines,
 			bundleStorage:       c.BundleStorage,
 			bundleDevices:       c.BundleDevices,
+			controllerName:      controllerName,
```
Comment (achilleasa) [inline cmd/juju/application/deploy.go]: There are other places that _partial_ bundleDeploySpecs are constructed. I would recommend populating the missing bits in `deployBundle` instead as I did for `TargetModelUUID` to be sure.

File: cmd/juju/application/deploy.go (hunk 2 above)
Comment (SimonRichardson) [reply cmd/juju/application/deploy.go]: I know right :| That took sometime to work out!

File: cmd/juju/application/deploy.go (hunk 3 above)
Comment (SimonRichardson) [reply cmd/juju/application/deploy.go]: Fair!

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**